source/internal/interactive_batch.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include "interactivity.h"
+#include "interactive_session.h"
+#include "rapidjson\document.h"
+#include <functional>
+
+namespace mixer_internal
+{
+
+struct interactive_batch_internal;
+struct interactive_batch_entry_internal;
+typedef std::function<void(interactive_batch_entry_internal*)> on_get_entry_params;
+typedef std::function<void(interactive_batch_internal*, rapidjson::Document::AllocatorType& allocator, rapidjson::Value& value)> on_get_batch_params;
+
+struct interactive_batch_internal
+{
+	interactive_session_internal* session;
+	const char* method;
```

Comment:
The respective batch begin method, e.g. https://github.com/mixer/interactive-cpp/pull/80/files/2337d46dacb92bc2621421564b7f6a5988c80a61#diff-426a8659af90d0fe57bec68619f83941R20

See [comment below](https://github.com/mixer/interactive-cpp/pull/80#discussion_r196289283) about modifying the document directly rather than needing to keep `param`.

Both strings need to be valid until `_commit` is called which is why they aren't copied.

Reply:
That will probably manifest in runtime errors where the caller cleans up the string before calling commit. Can we do a copy here and use std::string? It's a known runtime impact (exactly 2 string copies) and shouldn't be a perf issue.